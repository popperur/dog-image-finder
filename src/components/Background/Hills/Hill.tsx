import { HillHolder } from "components/Background/Hills/styles.tsx";

interface HillProps {
  bottom: number;
  left: number;
  width: number;
  height: number;
}

function Hill({ bottom, left, width, height }: HillProps) {
  const stylesArray = [
    `bottom: ${bottom}px`,
    `left: ${left}px`,
    `width: ${width}px`,
    `height: ${height}px`,
  ];

  return <HillHolder $styles={stylesArray} />;
}

export default Hill;
