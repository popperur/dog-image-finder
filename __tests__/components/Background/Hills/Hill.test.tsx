import "jest-styled-components";
import { render, RenderResult } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import Hill from "components/Background/Hills/Hill";

describe("Hill component", () => {
  let rendered: RenderResult;

  beforeEach(() => {
    rendered = render(<Hill bottom={0} left={20} width={40} height={60} />);
  });

  it("matches the snapshot", () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  it("adds the styles to the styled component", () => {
    const { container } = rendered;

    const hillHolder = container.firstChild;

    expect(hillHolder).toHaveStyleRule("bottom", "0px");
    expect(hillHolder).toHaveStyleRule("left", "20px");
    expect(hillHolder).toHaveStyleRule("width", "40px");
    expect(hillHolder).toHaveStyleRule("height", "60px");
  });
});
