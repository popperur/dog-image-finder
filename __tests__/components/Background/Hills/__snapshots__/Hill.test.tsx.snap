// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hill component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  background-color: #1f2731;
  border-top-left-radius: 100px;
  border-top-right-radius: 100px;
  transform: perspective(164px) rotatex(78deg) translate3d(0px, 18px, -42px);
  bottom: 0px;
  left: 20px;
  width: 40px;
  height: 60px;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
